build.gradle

```
@@ -15,97 +15,130 @@
  * along with Baritone.  If not, see <https://www.gnu.org/licenses/>.
  */
 
-group 'baritone'
-version '1.2.17'
+allprojects {
+    apply plugin: 'java'
+    apply plugin: "xyz.wagyourtail.unimined"
+    apply plugin: "maven-publish"
+
+    archivesBaseName = rootProject.archives_base_name
+    def vers = 'git describe --always --tags --dirty'.execute().text.trim()
+    if (!vers.startsWith("v")) {
+        println "git doesn't appear to be installed!"
+        println "using version number: " + rootProject.mod_version
+        version = rootProject.mod_version
```

Here's another attempt (I deleted the previous comment, don't search for it), this time actually making use of the fact that we can still use the commit hash even if we didn't find a tag.
```suggestion
    def vers;
    try {
        vers = 'git describe --always --tags --match=v* --dirty'.execute().text.trim()
    } catch (IOException ignored) {
        vers = ""
        println "git doesn't appear to be installed!"
    }
    if (vers.equals("")) {
        println "Couldn't find version tag!"
        println "Using fallback version number: " + rootProject.mod_version
        version = rootProject.mod_version
    } else if (!vers.startsWith("v")) {
        println "Couldn't find version tag!"
        println "Using fallback version number: " + rootProject.mod_version
        version = rootProject.mod_version + "-g" + vers
```